components/kyma-environment-broker/internal/process/provisioning/lms_cert_step.go

```diff
@@ -0,0 +1,187 @@
+	lmsTimeout               = 30 * time.Minute
+)
+
+type LmsClient interface {
+	RequestCertificate(tenantID string, subject pkix.Name) (id string, privateKey []byte, err error)
+	GetSignedCertificate(tenantID string, certID string) (cert string, found bool, err error)
+	GetCACertificate(tenantID string) (cert string, found bool, err error)
+	GetTenantStatus(tenantID string) (status lms.TenantStatus, err error)
+	GetTenantInfo(tenantID string) (status lms.TenantInfo, err error)
+}
+
+type lmsCertStep struct {
+	provider            LmsClient
+	repo                storage.Operations
+	normalizationRegexp *regexp.Regexp
+}
+
+func NewLmsCertificatesStep(certProvider LmsClient, repo storage.Operations) *lmsCertStep {
+
+	return &lmsCertStep{
+		provider:            certProvider,
+		repo:                repo,
+		normalizationRegexp: regexp.MustCompile("[^a-zA-Z0-9]+"),
+	}
+}
+
+func (s *lmsCertStep) Name() string {
+	return "Request_LMS_Certificates"
+}
+
+// Run executes getting LMS certificates steps, which means:
+// 1. check if the tenant is ready
+// 2. request certificates
+// 3. poll CA and signed certificates
+func (s *lmsCertStep) Run(operation internal.ProvisioningOperation, logger logrus.FieldLogger) (internal.ProvisioningOperation, time.Duration, error) {
+	if operation.Lms.Failed {
+		logger.Info("LMS has failed, skipping")
+		return operation, 0, nil
+	}
+
+	if operation.Lms.TenantID == "" {
+		logger.Error("Create LMS Tenant step must be run before")
+		return operation, 0, errors.New("the step needs to be run after 'Create LMS tenant' step")
+	}
+
+	// check if LMS tenant is ready
+	status, err := s.provider.GetTenantStatus(operation.Lms.TenantID)
+	if err != nil {
+		logger.Errorf("Unable to get LMS Tenant status: %s", err.Error())
+		if time.Since(operation.Lms.RequestedAt) > lmsTimeout {
+			logger.Error("Setting LMS operation failed - tenant provisioning timed out, last error: %s", err.Error())
+			return s.failLmsAndUpdate(operation)
+		}
+		return operation, tenantReadyRetryInterval, nil
+	}
+	if !(status.ElasticsearchDNSResolves && status.KibanaDNSResolves) {
+		logger.Infof("LMS tenant not ready: elasticDNS=%v, kibanaDNS=%v", status.ElasticsearchDNSResolves, status.KibanaDNSResolves)
+		if time.Since(operation.Lms.RequestedAt) > lmsTimeout {
+			logger.Error("Setting LMS operation failed - tenant provisioning timed out")
+			return s.failLmsAndUpdate(operation)
+		}
+		return operation, tenantReadyRetryInterval, nil
+	}
+
+	ti, err := s.provider.GetTenantInfo(operation.Lms.TenantID)
+	if err != nil {
+		logger.Errorf("Unable to get LMS Tenant info: %s", err.Error())
+		if time.Since(operation.Lms.RequestedAt) > lmsTimeout {
+			logger.Error("Setting LMS operation failed - tenant provisioning timed out, last error: %s", err.Error())
+			return s.failLmsAndUpdate(operation)
+		}
+		return operation, tenantReadyRetryInterval, nil
+	}
+	host := ti.DNS
+
+	// request certificates
+	pp, err := operation.GetProvisioningParameters()
```

Comment:
I know this step is not first, but I suggest make `GetProvisioningParameters` first, because if parameters are wrong then all calls to LmsClient are useless :) step one way or another will fail